src/main/java/com/uid2/admin/vertx/service/ServiceLinkService.java

```diff
@@ -92,6 +93,7 @@ private void handleServiceLinkAdd(RoutingContext rc) {
             Integer serviceId = body.getInteger("service_id");
             Integer siteId = body.getInteger("site_id");
             String name = body.getString("name");
+            JsonArray rolesJson = body.getJsonArray("roles");
             if (linkId == null || serviceId == null || siteId == null || name == null) {
```

Comment:
It looks like empty names are allowed here. Is that intentional?

Reply:
No, good catch! I'll add that in.